mbed_host_tests/__init__.py

```diff
@@ -282,6 +282,10 @@ def init_host_test_cli_params():
                       dest='global_resource_mgr',
                       help='[Experimental] Global resource manager service module name, IP and port, example remote_client:10.2.123.43:3334')
 
+    parser.add_option('-s', '--srm',
+                      dest='simulator_resource_mgr',
+                      help='[Experimental] Simulator resource manager service module name, and a config file, example fastmodel_agent:DEFAULT')
```

Comment:
Since the srm implements the same connector primitive as the grm, could we not merge the two arguments? Does it mostly come down to the fact that the srm doesn't require the IP address and port? If so, we could probably relax that requirement of the grm.

Reply:
@bridadan I initially thought about using the same --grm option. But after working on it for a while, I realized it is not very practical.

Here  are some of my concerns:
1. The required parameters are different: --grm need a server IP and port number to the remote server. --srm just require a config_file.
2.  The internal logic of connection is different: --grm will need query the server for an available target. if the target is used by others, it will wait for it been released. --srm not having this logic at all. 
3. the target behavior is different: --grm still use the real hardware via serial, and need to flash and reset the target. --srm will need to launch the model first before connecting to it and use model-specific load/run function, also serial in replaced via socket. 

So I can image, even we can get them merged into the same primitive connector, most of the code will use lost of if/else redirect them to a totally different route.
And not mention that will require both code change from htrun and RaaS-pyclient, which will be risky for broken CI test. (As I know lots of CI using --grm interface, and the command hard-coded in the Jenkins side)

Reply:
Thanks for raising those points, I'll go through them a bit:

1. I think we could probably relax this requirement on the `--grm` argument and leave it to the module to define their own arguments. This wouldn't be a breaking change for the existing modules in use and it could open the door for the fastmodel_agent.

2. While the internal logic is different, I don't think that is all that important from the perspective of the user. You can supply anything as a "global resource manager", including simulators.

3. I'd say probably the same thing I said in `#2`. What is actually happening isn't really important for the use to know. If they are supplying the module on the command line, then I think we can expect that they are familiar with what is happening on the other end 😄 

And for your last point:
> And not mention that will require both code change from htrun and RaaS-pyclient, which will be risky for broken CI test

I don't think it'll require any changes from the raas-pyclient, just changes on the htrun side. And we can write some test cases ahead of time to be sure we don't break it. The interface is relatively simple so I think should be quite doable.

I realize this is all extending the integration time for your component here, is this time sensitive?

Reply:
I see your point.
So do you mean change both the connection types share the same --grm command line option, but use different primitive connectors? I think that's possible, and fairly easy change.

Currently, the function for parsing --grm option is in htrun : https://github.com/ARMmbed/htrun/blob/master/mbed_host_tests/host_tests_runner/host_test_default.py#L180-L188
If we want to 
> leave it to the module to define their own arguments.

we have to move such function into RaaS_clinet or fastmodel_agent module.
That why I said it will need code changes into RaaS_client.

Otherwise, we will need leave this in the htrun, which means determin the arguments depens on raas_client or fastmodel_agent

Reply:
@bridadan for timing wise.
Actually I don't know how is the testing tools keep up with mbed-os release.
I would really like to get this merged before 5.9.

I didn't any tags on htrun.
which I guess it is going to be a bit more flexable compare to mbed-os, which will need to have the code freeze by the end of next week for 5.9?

Reply:
>we have to move such function into RaaS_clinet or fastmodel_agent module.
That why I said it will need code changes into RaaS_client.

Oh duh, you're right! Yeah so maybe that's a bit more out of scope than I thought. I'd like to hear from @theotherjimmy before we make any definite decisions on the path forward.

Yeah for releases, htrun will typically trail Mbed OS a little bit, so we can always release this at 5.9 or a little after.

Reply:
The differing parameters are the thing that gets it for me. 2 methods of doing the same thing are bad if they take the same parameters, 2 implementation of an interface are good when they're constructed from differing params.

Reply:
`--srm` is okay. Maybe we should consider a different switch name?

Reply:
I don't actually have a problem with the name. And I've thought a bit more about the current implementation and it really seems ok as is. I was kind of worried that we were duplicating stuff, but that's something we could always address in the future.

Reply:
Hi guys @theotherjimmy @bridadan,
Is there any conclusion on this PR?

Do you think is the switch "-s, --srm" appropriate ?
current usage is *--srm <module_name>:<config_name>*
*e.g. --srm fm_agent:DEFAULT*
I started design it like this, becasue I thought other people could like write other simulator module for their own models.

Now I start to favor a new switch without module_name, module_name is hard coded inside htrun
*e.g. --fm <config_name>*

Reply:
BTW. what's the error of circleci about?
It seems passed on my laptop.

Reply:
I'd be fine with moving to your proposed `--fm`. Makes it a little simpler!

For the circleci failure, I don't think that's relevant to your changes.